torchao/float8/float8_tensor.py

```diff
@@ -313,7 +313,7 @@ def __new__(
             linear_mm_config if linear_mm_config is not None else LinearMMConfig()
         )
         self._gemm_input_role = gemm_input_role
-        assert axiswise_dim in (None, 0, -1), f"unsupported axiswise_dim {axiswise_dim}"
+        assert axiswise_dim in (None, 0, 1, -1), f"unsupported axiswise_dim {axiswise_dim}"
```

Comment:
Note for reviewer: I had to do this to handle the 3D B tensor, which the user will pass in transposed (e.g. `_scaled_grouped_mm(A, B.transpose(-2, -1), ...)`). The rowwise scales for a 3D transposed tensor need to be along dim 1, which was not supported. 

Example:

B.shape = (B,N,K)
B.transpose(-2,-1).shape = (B,K,N)
B_scales need to be (B,1,N) and then be squeezed into (B,N) to prepare for torch._scaled_grouped_mm input